NuKeeper.Inspection/RepositoryInspection/PaketRepositoryScanner.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using System.Collections.Generic;
+using Microsoft.FSharp.Core;
+using NuKeeper.Abstractions.Inspections.Files;
+using Paket;
+
+namespace NuKeeper.Inspection.RepositoryInspection
+{
+    public class PaketRepositoryScanner : IRepositoryScanner
+    {
+        private readonly IReadOnlyCollection<IPackageReferenceFinder> _finders;
+
+        public PaketRepositoryScanner(ProjectFileReader projectFileReader, PackagesFileReader packagesFileReader,
+            NuspecFileReader nuspecFileReader, DirectoryBuildTargetsReader directoryBuildTargetsReader)
+        {
+            _finders = new IPackageReferenceFinder[]
+                {projectFileReader, packagesFileReader, nuspecFileReader, directoryBuildTargetsReader};
+        }
+
+        public IReadOnlyCollection<PackageInProject> FindAllNuGetPackages(IFolder workingFolder)
+        {
+            var deps = new Dependencies($"{workingFolder.FullPath}/paket.dependencies");
+            try
+            {
+                var outdated = deps.FindOutdated(false, false, false, FSharpOption<string>.None); //todo see if we can get the current version list immeditaily
```

Comment:
If we are looking to get all packages for NuKeeper to do the outdated logic, then we could use [GetDirectDependencies](https://github.com/fsprojects/Paket/blob/702432a917957e28ebb303541796c86f19e1ab46/src/Paket.Core/PublicAPI.fs#L563)